src/components/appNavigation/stackNavigation.js

```diff
@@ -57,8 +57,13 @@ class AppView extends Component<AppViewProps, AppViewState> {
    */
   trans: boolean = false
 
-  shouldComponentUpdate() {
-    return this.trans === false
+  shouldComponentUpdate(nextProps, nextState) {
+    const navStateChanged =
+      isEqualWith(nextProps, this.props, (val1, val2) => (isFunction(val1) && isFunction(val2) ? true : undefined)) ===
```

Comment:
i don't understand that logic - tell me please why this is needed, which issue it solves ?
probably you'd need to just memoize some callback passed to this components via useCallback instead of using hacks like that ?

Reply:
yes this prevent lots of re-rendering that happens in stacknavigation. (see FIXME logs)
specifically it causes the "TabView" to also be rerendered many times and it causes some flashing of the label with numbers I show. (updateIcon method)
the re-rendering comes from react-navigation I believe, i couldnt find the exact source for it
this is the solution I have.

Reply:
ok